projects/storefrontlib/src/lib/store-finder/services/store-finder.service.ts

```diff
@@ -8,43 +8,57 @@ import * as fromStore from '../store';
 
 @Injectable()
 export class StoreFinderService {
+  private geolocationWatchId: number;
+
   constructor(
     private store: Store<fromStore.StoresState>,
     private winRef: WindowRef
   ) {}
 
   findStores(queryText: string, useMyLocation?: boolean) {
     if (useMyLocation) {
-      this.store.dispatch(new fromStore.OnHold());
-      this.winRef.nativeWindow.navigator.geolocation.getCurrentPosition(
+      this.clearWatchGeolocationAndDispatchAction(new fromStore.OnHold());
+      this.geolocationWatchId = this.winRef.nativeWindow.navigator.geolocation.watchPosition(
         (pos: Position) => {
           const longitudeLatitude: LongitudeLatitude = {
             longitude: pos.coords.longitude,
             latitude: pos.coords.latitude
           };
-          this.store.dispatch(
+          this.clearWatchGeolocationAndDispatchAction(
```

Comment:
Well, I am not sure about the name. I guess that `clearWatchGeolocation(callbackAction: function)` is enough